### DISCUSSION TITLE
How to use git LFS on windows on private repo over SSH?

### DISCUSSION THREAD

**Discussion by stephane-archer** [community/community#133261] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

```
 git lfs push --all origin
Uploading LFS objects:   0% (0/3), 0 B | 0 B/s, done.
batch request: git@github.com: Permission denied (publickey).: exit status 255
```
I didn't had to enter my passphrase so I think it doesn't use my ssh key to push LFS files

**Comment by Rakhshan7070** [comment]
I am getting the same error on linux mint, also tried some fixes but none worked, here's a discussion regarding the same issue 
https://github.com/git-lfs/git-lfs/discussions/4848

**Comment by divyamlathiya** [comment]
Hey,

It looks like your SSH key isn’t being used when pushing LFS files, which is why you’re not prompted for your passphrase. Here's what you can try:

1. **Check if your SSH key is added:**
Run this to see if your SSH key is loaded:

```
ssh-add -l
```

If it’s empty, you’ll need to add your key with:

```
ssh-add ~/.ssh/id_rsa
```

2. **Check your remote URL:**
Make sure your remote URL is using SSH and not HTTPS. You can check it with:

```
git remote -v
```

If it's using HTTPS (e.g., `https://github.com/username/repo.git`), change it to SSH:

```
git remote set-url origin git@github.com:username/repo.git
```

3. **Verify SSH key on GitHub:**
Double-check that your SSH key is added to your GitHub account. Go to **GitHub Settings > SSH and GPG** keys and make sure your public key is listed there.

4. **Test the SSH connection:**
Run this command to see if SSH is working properly:

```
ssh -T git@github.com
```

You should see a message like:
"Hi username! You've successfully authenticated, but GitHub does not provide shell access."

5. **Try pushing again:**
After making sure everything is set up, try your push again:

```
git lfs push --all origin
```

Hopefully, that gets it working! Let me know if you're still stuck.

**Reply by Rakhshan7070** [reply]
> Hey,
> 
> It looks like your SSH key isn’t being used when pushing LFS files, which is why you’re not prompted for your passphrase. Here's what you can try:
> 
> 1. **Check if your SSH key is added:**
>    Run this to see if your SSH key is loaded:
> 
> ```
> ssh-add -l
> ```
> 
> If it’s empty, you’ll need to add your key with:
> 
> ```
> ssh-add ~/.ssh/id_rsa
> ```
> 
> 2. **Check your remote URL:**
>    Make sure your remote URL is using SSH and not HTTPS. You can check it with:
> 
> ```
> git remote -v
> ```
> 
> If it's using HTTPS (e.g., `https://github.com/username/repo.git`), change it to SSH:
> 
> ```
> git remote set-url origin git@github.com:username/repo.git
> ```
> 
> 3. **Verify SSH key on GitHub:**
>    Double-check that your SSH key is added to your GitHub account. Go to **GitHub Settings > SSH and GPG** keys and make sure your public key is listed there.
> 4. **Test the SSH connection:**
>    Run this command to see if SSH is working properly:
> 
> ```
> ssh -T git@github.com
> ```
> 
> You should see a message like: "Hi username! You've successfully authenticated, but GitHub does not provide shell access."
> 
> 5. **Try pushing again:**
>    After making sure everything is set up, try your push again:
> 
> ```
> git lfs push --all origin
> ```
> 
> Hopefully, that gets it working! Let me know if you're still stuck.

Thanks for the info but it wasn't the issue, ssh was being used, actually the  thing was that the .lfsconfig was set to a custom organisation where I didn't had write permission, to fix that I just nuked .lfsconfig and it worked.
You can have a look here: https://github.com/crdroidandroid/proprietary_vendor_motorola_dubai/commit/e9ea8f73d3829db3e363f13250ed93831f62296c
